tls/s2n_config.c

```diff
@@ -95,6 +95,7 @@ static int s2n_config_init(struct s2n_config *config)
     config->encrypt_decrypt_key_lifetime_in_nanos = S2N_TICKET_ENCRYPT_DECRYPT_KEY_LIFETIME_IN_NANOS;
     config->decrypt_key_lifetime_in_nanos = S2N_TICKET_DECRYPT_KEY_LIFETIME_IN_NANOS;
     config->async_pkey_validation_mode = S2N_ASYNC_PKEY_VALIDATION_FAST;
+    config->validate_x509_time = true;
 
```

Comment:
This seems unnecessarily risky. We shouldn't be disabling timestamp validation by default. I know it makes the name worse, but the flag should really provide the safe behavior when zero-inited.

More name suggestions: "disable_x509_time_validation" "disable_x509_timestamp_checks" "disable_x509_validation_of_timestamps"